www/src/components/IconsTable.jsx

```diff
@@ -1,27 +1,177 @@
+
+  const handleClick = () => {
+    setCurrentIcon(iconName);
+    if (previewRef.current) {
+      previewRef.current.scrollIntoView({ behavior: 'smooth', block: 'center' });
+    }
+  };
+
+  if (!iconName) {
+    return null;
+  }
+
+  return (
+    <>
+      <div
+        role="button"
+        key={iconName}
+        className="pgn-doc__icons-table__cell"
+        style={style}
+        onKeyPress={handleClick}
+        onClick={handleClick}
+        tabIndex={0}
+      >
+        <Icon
+          key={iconName}
+          src={IconComponents[iconName]}
+        />
+        <span className="pgn-doc__icons-table__cell-text">{iconName}</span>
+      </div>
+    </>
+  );
+};
+
+const IconsTable = () => {
+  const previewRef = React.useRef(null);
+  const [searchValue, setSearchValue] = useState('');
+  const [iconsList, setIconsList] = useState(ICON_NAMES);
+  const [currentIcon, setCurrentIcon] = useState(ICON_NAMES[0]);
+  const [showToast, setShowToast] = useState(false);
+  const currentIconImport = `import { ${currentIcon} } from '@edx/paragon/icons';`;
+
+  const copyToClipboard = (content) => {
+    navigator.clipboard.writeText(content);
+    setShowToast(true);
+  };
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const debouncedSetIconsList = useCallback(
+    debounce(setIconsList, 250, { leading: true }),
+    [],
+  );
+
+  useEffect(() => {
+    debouncedSetIconsList(
+      ICON_NAMES.filter(name => name.toLowerCase().includes(searchValue.toLowerCase())),
+    );
+  }, [searchValue, iconsList, debouncedSetIconsList]);
+
   return (
-    <div>
-      {Object.keys(IconComponents).map(iconName => (
-        <div key={iconName} className="d-flex mb-3 align-items-center">
+    <>
+      <SearchField
+        onSubmit={() => {}}
+        onChange={value => setSearchValue(value)}
+        placeholder="Search icons"
+      />
+      {currentIcon && (
+        <div className="pgn-doc__icons-table__preview" ref={previewRef}>
+          <h3
+            role="presentation"
+            className="rounded"
+            onClick={() => copyToClipboard(currentIcon)}
+          >
+            {currentIcon}
+          </h3>
           <Icon
-            className="mr-3"
-            key={iconName}
-            src={IconComponents[iconName]}
+            key={currentIcon}
+            src={IconComponents[currentIcon]}
           />
-          <h3 className="mb-0 mr-3 flex-grow-1 h5">{iconName}</h3>
-          <code className="bg-light px-2 py-1 rounded">
+          <code
+            role="presentation"
+            className="rounded"
+            onClick={() => copyToClipboard(currentIconImport)}
+          >
             <small>
-              {`import { ${iconName} } from '@edx/paragon/icons';`}
+              {currentIconImport}
             </small>
           </code>
         </div>
-      ))}
-    </div>
+      )}
+      {iconsList.length ? (
+        <div className="pgn-doc__icons-table">
+          <AutoSizer>
```

Comment:
It appears there's a double-scroll situation in which this table can be scrolled as well as the main page viewport. Is there a way we can eliminate the need for the table to be scrollable?

If we're trying to eliminate displaying icons that aren't visible in the viewport, I'd think additional rows of icons would start appearing as the main page scroll gets closer to the end of the page, if that makes sense.

Reply:
I changed to own infinite scroll implementation